### PR TITLE
Added all the API(back-end) codes.

### PR DESCRIPTION
**Pull request by md-oshim**

### CODE REVIEW

File: API/RepositoryLayer/IRepository.cs
Comment (Rreayz) [inline API/RepositoryLayer/IRepository.cs:1]: Why GetAllAsync() returns IAsyncEnumerable and GeneralSearch() return IList? Make them similar to each other, to keep the consistency.

File: API/RepositoryLayer/Repository.cs
Comment (Rreayz) [inline API/RepositoryLayer/Repository.cs:1]: Why are some functions are async and some others as sync methods? Make all function async method, call them with await keyword.

File: API/ServiceLayer/Implementations/TouristPlaceServices.cs
Comment (Rreayz) [inline API/ServiceLayer/Implementations/TouristPlaceServices.cs:1]: Make all functions as async method.

File: API/TouristPlaceWebApi/Controllers/TouristPlaceController.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+        [HttpGet(nameof(GetAllTouristPlaces))]
+        public async Task<ActionResult<List<TouristPlace>>> GetAllTouristPlaces(string? searchedText, string? sortBy, string? sortType)
+        {
+            List<TouristPlace> allTouristPlaces = new();
+            // searching mechanism
+            if(searchedText.IsNullOrEmpty() == false)
+            {
+                searchedText = searchedText.Trim();
+                allTouristPlaces = (List<TouristPlace>)_touristPlaceServices.TouristPlaceGeneralSearch(searchedText);
```
Comment (Rreayz) [inline API/TouristPlaceWebApi/Controllers/TouristPlaceController.cs]: call all database methods with await keyword.

File: API/RepositoryLayer/IRepository.cs
Comment (md-oshim) [reply API/RepositoryLayer/IRepository.cs:1]: Vaiya, as you told to make the assignment from synchronous to asynchronous. I converted all the synchronous methods into asynchronous but for implementing the GeneralSearch I had to use Where method of EF Core which doesn't support asynchronous functionality. This is why I could not convert it to asynchronous method. Hence GetAllAsync() returns IAsyncEnumerable and GeneralSearch() return IList.

File: API/RepositoryLayer/Repository.cs
Comment (md-oshim) [reply API/RepositoryLayer/Repository.cs:1]: okay vaiya

File: API/ServiceLayer/Implementations/TouristPlaceServices.cs
Comment (md-oshim) [reply API/ServiceLayer/Implementations/TouristPlaceServices.cs:1]: Vaiya, including TouristPlaceGeneralSearch?

Comment (Rreayz) [reply API/ServiceLayer/Implementations/TouristPlaceServices.cs:1]: yes

**Review by Rreayz — APPROVED** [review]

**Pull request merged**